### DISCUSSION TITLE
Idea for new Focus Mode: dpad_and_keyboard

### DISCUSSION THREAD

**Discussion by danideter** [godotengine/godot-proposals#8832] [GUI]
Hi there,

I'm making an app primarily for mobile that I might like to port to PC someday. For a quiz minigame, if the user is using touch or a mouse, it makes little sense to grab the focus of the last selected button. However, if the user is useing a keyboard or a dpad, focus helps the player know which button they are about to press:

![focus_mode](https://github.com/godotengine/godot-proposals/assets/13397972/d8ea72bb-3c06-434f-8674-4694f9ddb4c6)

Currently, the focus modes are:
None
Click
All

Where all includes click, dpad, and keyboard. Would it be useful to others to provide a 4th option: dpad_and_keyboard?

I was hoping it would be a relatively simple change, and tried to make a PR. Unfortunately, I couldn't find where the logic for the click modes are implmented. I did find the following files which would likely need to be updated:
https://github.com/godotengine/godot/blob/c8c483cf57a768110fce57e509f9b855e69d34b7/doc/classes/Control.xml#L1140
https://github.com/godotengine/godot/blob/c8c483cf57a768110fce57e509f9b855e69d34b7/scene/gui/control.cpp#L3607
https://github.com/godotengine/godot/blob/c8c483cf57a768110fce57e509f9b855e69d34b7/scene/gui/control.h#L62

**Comment by Calinou** [comment]
> For a quiz minigame, if the user is using touch or a mouse, it makes little sense to grab the focus of the last selected button

I think we need to handle this differently. We should make it so mouse-produced focus doesn't have a visible outline by default, as I proposed in https://github.com/godotengine/godot-proposals/issues/6577.

This way, you still preserve expected accessibility behavior (clicking with the mouse defines focus), but it doesn't visually get in the way.

**Reply by danideter** [reply]
Ah yes, that would work too :)
